python/hail/expr/expressions/expression_typecheck.py

```diff
@@ -53,6 +53,7 @@ def can_coerce(self, t: HailType) -> bool:
         ...
 
     def coerce(self, x: Expression) -> Expression:
+        x = to_expr(x)
```

Comment:
why do we call to_expr if x is said to have type Expression?

Reply:
I guess it doesn't have to be an expression. I'll remove the type hint.